docling_core/experimental/serializer/doctags.py

```diff
@@ -0,0 +1,418 @@
+        my_visited = visited or set()
+        parts = doc_serializer.get_parts(
+            node=item,
+            list_level=list_level,
+            is_inline_scope=True,
+            visited=my_visited,
+            **kwargs,
+        )
+        wrap_tag = DocumentToken.INLINE.value
+        text_res = "\n".join([p.text for p in parts if p.text])
+        text_res = f"{text_res}\n"  # NOTE: explicit
+        text_res = _wrap(text=text_res, wrap_tag=wrap_tag)
+        return SerializationResult(text=text_res)
+
+
+class DocTagsFallbackSerializer(BaseFallbackSerializer):
+    """DocTags-specific fallback serializer."""
+
+    @override
+    def serialize(
+        self,
+        *,
+        item: NodeItem,
+        doc_serializer: "BaseDocSerializer",
+        doc: DoclingDocument,
+        **kwargs,
+    ) -> SerializationResult:
+        """Serializes the passed item."""
+        if isinstance(item, DocItem):
+            text_res = "<!-- missing-text -->"
+        else:
+            text_res = ""  # TODO go with explicit None return type?
+        return SerializationResult(text=text_res)
+
+
+class DocTagsDocSerializer(DocSerializer):
+    """DocTags-specific document serializer."""
+
+    text_serializer: BaseTextSerializer = DocTagsTextSerializer()
+    table_serializer: BaseTableSerializer = DocTagsTableSerializer()
+    picture_serializer: BasePictureSerializer = DocTagsPictureSerializer()
+    key_value_serializer: BaseKeyValueSerializer = DocTagsKeyValueSerializer()
+    form_serializer: BaseFormSerializer = DocTagsFormSerializer()
+    fallback_serializer: BaseFallbackSerializer = DocTagsFallbackSerializer()
+
+    list_serializer: BaseListSerializer = DocTagsListSerializer()
+    inline_serializer: BaseInlineSerializer = DocTagsInlineSerializer()
+
+    params: DocTagsParams = DocTagsParams()
+
+    @computed_field  # type: ignore[misc]
+    @cached_property
+    def _kwargs(self) -> dict[str, Any]:
+        return self.params.model_dump()
+
+    def get_params(self) -> dict[str, Any]:
+        """Get parameters for serialization."""
+        return self._kwargs
+
+    def post_process(
+        self,
+        text: str,
+        *,
+        escape_html: bool = True,
+        formatting: Optional[Formatting] = None,
+        hyperlink: Optional[Union[AnyUrl, Path]] = None,
+        **kwargs,
+    ) -> str:
+        """Apply some text post-processing steps."""
+        res = text
+        if escape_html:
+            res = html.escape(res, quote=False)
+        res = super().post_process(
+            text=res,
+            formatting=formatting,
+            hyperlink=hyperlink,
+        )
+        return res
+
+    @override
+    def serialize(self, **kwargs) -> SerializationResult:
+        """Run the serialization."""
+        parts = self.get_parts()
+        content = "\n".join([p.text for p in parts if p.text])
+        wrap_tag = DocumentToken.DOCUMENT.value
+        text_res = f"<{wrap_tag}>{content}\n</{wrap_tag}>"
```

Comment:
Here, the newline might become optional

Reply:
Additional whitespace can now be completely prevented via the new "minified" mode.